core/src/main/scala/akka/kafka/ProducerMessage.scala

```diff
@@ -27,18 +33,62 @@ object ProducerMessage {
   final case class Message[K, V, +PassThrough](
       record: ProducerRecord[K, V],
       passThrough: PassThrough
-  )
+  ) extends Messages[K, V, PassThrough] {
+
+    /**
+     * Returns this Message if applying the predicate to
+     * this Message returns true. Otherwise, returns a [[PassThroughMessage]].
+     */
+    def filter(p: Message[K, V, PassThrough] => Boolean): Messages[K, V, PassThrough] =
```

Comment:
This exposes scala function to the Java API as well. However it might be not bad as it is a SAM and can be used from java code like:
```java
.map(msg -> msg.filter(m -> m.record().key().equals("key")))
```

Reply:
Question: We still cross build for Scala 2.11.  Even though the 2.11 artifact is built with the 1.8 JVM can it still be used by Java 1.7 users?

Reply:
No, the class files are in JDK 8 format.

Reply:
I think you can say that Java users should use `_2.12` artifacts. For Akka we tend to still support `_2.11` for Java, but there shouldn't be any strong reasons why one would use that.